bears/vcs/git/CISkipInspectBear.py

```diff
@@ -0,0 +1,63 @@
+import re
+
+from bears.vcs.git.GitCommitMetadataBear import GitCommitMetadataBear
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.results.Result import Result
+
+
+class CISkipInspectBear(GlobalBear):
+    LANGUAGES = {'Git'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+
+    BEAR_DEPS = {GitCommitMetadataBear}
+
+    def run(self, dependency_results, filename_patterns: list,
+            appveyor_ci: bool = False, **kwargs):
+        """
+        Check the HEAD commit to see if it disables CI build and
+        return result accordingly. Supported CI include AppVeyor,
+        Bitrise, Circle CI, GitLab CI, Scrutinizer, Semaphore,
+        Shippable, Travis CI and wercker.
+
+        :param filename_patterns:   A list of regex to identify files
+                                    that should not skip CI builds.
+        :param appveyor_ci:         Whether AppVeyor is used by the
+                                    project or not.
+        """
+        for result in dependency_results[GitCommitMetadataBear.name]:
+
+            supported_regex = r'\[ci skip\]|\[skip ci\]'
+
+            # AppVeyor allows to skip build by commit if the
+            # supported sequence is present in the title of the
+            # commit only. All other CI tools allow these sequences
+            # to be present anywhere in the commit message.
+            if appveyor_ci:
+                supported_regex += (r'|\[skip appveyor\]')
+                pos = result.raw_commit_message.find('\n')
+                commit_title = result.raw_commit_message[:pos] if (
+                    pos != -1) else result.raw_commit_message
+
+                match = re.search(supported_regex, commit_title)
+
+            else:
+                match = re.search(
+                    supported_regex, result.raw_commit_message)
+
+            if match:
+
+                all_files = (result.modified_files +
+                             result.added_files + result.deleted_files)
+
+                for file in all_files:
+                    for pattern in filename_patterns:
+                        if re.search(pattern, file):
+                            yield Result(
+                                self,
+                                'This commit modifies a file that has '
+                                'pattern of type "%s", thus should '
+                                'not disable CI build.' %
+                                pattern)
+                            return
```

Comment:
why `return` ?  there is a loop going on, and the bear should process all data it is being given.